Implemented an 'afterExtract' function hook  RE: #270

MarshallOfSound opened the pull request:
This PR adds an `afterExtract` option to the API.  This option should be an array of functions, each of which is called directly after `electron-packager` has finished extracting to the temporary directory.

Relevant issue: #270

malept commented on index.js:
```diff
@@ -125,7 +125,16 @@ function createSeries (opts, archs, platforms) {
+                  })
+                  series(newFunctions, cb)
+                }
+              })
```
I think the callback you defined for `extract` be another function in the series.

MarshallOfSound replied:
Are you suggesting then that we should do

```js
series([extract(zipPath, {dir: buildDir})].concat(newFunctions))
```

malept replied:
I was thinking more along the lines of

```javascript
function (cb) {
  extract(zipPath, {dir: buildDir}, cb)
},
function (cb) {
  if (opts.afterExtract exists and is not array) {
    return error
  }
  // etc.
}
```
...that got pseudocode-y, fast.

MarshallOfSound replied:
Ah got it :+1: 

Done

malept commented on docs/api.md:
```diff
@@ -45,6 +45,16 @@ The non-`all` values correspond to the platform names used by [Electron releases
 
+##### `afterExtract`
+
+*Array*
```
Should probably read

> *Array* of *Function*s

malept commented on docs/api.md:
```diff
@@ -45,6 +45,16 @@ The non-`all` values correspond to the platform names used by [Electron releases
+
+* `buildPath` - The path to the temporary folder where Electron has been extracted to
+* `electronVersion` - The version of electron you are packaging for
+* `callback` - Must be called once you have completed your actions
```
Minor nit: the rest of the file has standardized on `-` for list items.

malept commented on docs/api.md:
```diff
@@ -45,6 +45,18 @@ The non-`all` values correspond to the platform names used by [Electron releases
+
+*Array*
+
+An array of functions to be called after Electron has been extracted to a temporary directory.  Each function is called with three parameters:
```
> five parameters

*(Insert Spanish Inquisition meme here)*

MarshallOfSound replied:
My counting skillz are amazing :+1:

malept commented:
Looks good, I think this just needs at least one unit test.

MarshallOfSound commented:
@malept Not too familiar with the testing framework here but I added some tests :+1:

malept commented:
The test logic looks OK at a glance (I'll look at it in more detail tomorrow), but I'd prefer if it had its own test instead of tacking it onto another test.

MarshallOfSound commented:
@malept I've moved the logic to it's own test and moved it to a new file called "hooks" so that if in the future we add more of these hooks we have a place for them to go :+1:

malept commented:
Sorry about the delay, work & real life got in the way.

I had merged #351 before this one, so if you could take care of the `strict` changes in `test/hooks.js`, I'll merge.

MarshallOfSound commented:
@malept Talk about delay 😆 

I finally got around to adding `use strict` 😁

malept commented:
Amusingly enough, this morning I did something similar. The only difference is that the require statements are `const`:
https://github.com/electron-userland/electron-packager/tree/after-extract-hook

MarshallOfSound commented:
@malept So who's do we use 😆

malept commented:
I added a PR for my branch (#403). I'll wait for CI to run and then merge it.
